tests/scripts/test_market_data_ingestion.py

```
@@ -1165,6 +1221,396 @@ def fake_load(df, metadata=None):
+        ("TWSE", "parse", "TWSE batch parse failed (ValueError)."),
+        ("TPEX", "fetch", "TPEX batch fetch failed (HTTPError)."),
+        ("TPEX", "parse", "TPEX batch parse failed (ValueError)."),
+    ],
+)
+def test_market_batch_failures_do_not_expose_exception_details(
+    exchange,
+    expected_message,
+    monkeypatch,
+    stage,
+):
+    feed_url_with_token = "https://feed.test/market?token=secret"
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to: "feed_url_with_token"

(S105)